### PR TITLE
Drop Python 3.9 Support

### LINKED ISSUE #553: Drop Python 3.9 Support
**Describe the feature you'd like added to this project**
Anti-feature: drop Python 3.9 Support in the Spring (when Xarray drops Python 3.9) to be [SPEC 0](https://scientific-python.org/specs/spec-0000/) compliant

### PR DESCRIPTION
**Pull request by cyschneck**
## PR Summary

Closes #553

## PR Checklist

**General**
- [x] Make an issue if one doesn't already exist
- [x] Link the issue this PR resolves by adding `closes #XXX` to the PR description where XXX is the number of the issue.
- [x] Add a brief summary of changes to `docs/release-notes.rst` in a relevant section for the next unreleased release. Possible sections include: Documentation, New Features, Bug Fixes, Internal Changes, Breaking Changes/Deprecated
- [x] Add appropriate labels to this PR
- [x] Make your changes in a forked repository rather than directly in this repo
- [x] Open this PR as a draft if it is not ready for review
- [x] Convert this PR from a draft to a full PR before requesting reviewers
- [x] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.
- [x] If needed, squash and merge PR commits into a single commit to clean up commit history

### CODE REVIEW

**Comment by cyschneck** [comment]
The current PR checks are expecting the three remaining tests for Python 3.9 across platforms, but they are no longer being generating, so those three unfinished tests can be ignored

**Review by anissa111 — APPROVED** [review]
Looks good, thank you!

I'll go turn off the required checks for 3.9 now.

File: docs/release-notes.rst [hunk 1]
```diff
@@ -11,8 +11,10 @@ Upcoming release
 
 Internal Changes
 ^^^^^^^^^^^^^^^^
+* Drop Python 3.9 Support by `Cora Schneck`_ in (:pr:`599`)
```
Comment (kafitzgerald) [inline docs/release-notes.rst:14]: I think we probably want this in a different category.  Maybe "Compatibility Notes" or "Breaking Changes"?  @anissa111, I've been meaning to ask - is there a list of categories you use for release notes?

**Review by kafitzgerald — CHANGES_REQUESTED** [review]
Looks good other than the minor release notes item!

I am curious if we have any pending or planned work that's not 3.9 compatible.  

If not, I might have a slight preference for holding off a little bit on this (at least until the end of the iteration).  It looks like xarray [hasn't dropped 3.9 quite yet](https://github.com/pydata/xarray/pull/8937) and our examples gallery can't move to 3.12 yet because of wrf-python.

**Pull request merged**